### PR TITLE
[WIP] Add basic VXI11 support

### PR DESCRIPTION
**Pull request by zaporozhets**
Yo Guys! Could you please take a look at my basic implementation of VXI11 example for parser.

Cheers,
Taras

### CODE REVIEW

**Comment by jancumps** [comment]
Can this be conditionally included/excluded in the compilation? Asking because I'm using the library in a memory challenged microcontroller.

**Comment by zaporozhets** [comment]
Yes, libscpi can be compiled without vxi11 support.

File: examples/test-vxi11/main.c [hunk 1]
```diff
@@ -0,0 +1,632 @@
...
+
+#define RX_BUF_SIZE 512
+
+/* a global output buffer to collect output data */
+#define SCPI_OUPUT_BUFFER_SIZE (256)
+char SCPI_outputBuffer[SCPI_OUPUT_BUFFER_SIZE];
+unsigned int SCPI_outputBuffer_wr_idx = 0;
+unsigned int SCPI_outputBuffer_rd_idx = 0;
+
+char available_bytes = 0;
```
Comment (zaporozhets) [inline examples/test-vxi11/main.c]: TODO: unused

**Comment by j123b567** [comment]
Thank you for this example of VXI11 usage.

I'm sorry for so late response. I thought that I can review it and test it but I was unable to do it until now.

Please, do you have the source file for `rpcgen` which was used to generate `vxi11_xdr.c`?
Do you have some hints for me, how can I try this? I don't have any experience with VXI11. e.g. how  to generate VXI11 counterpart so I can talk with this example.

**Comment by pigrew** [comment]
(I'm lurking around, watching this project, though I don't have any intentions of using this patch at this moment. On the other hand, I figure I can answer your questions.)

> Please, do you have the source file for `rpcgen` which was used to generate `vxi11_xdr.c`?

`rpcgen.c` comes with from old SUN UNIX utilities. There are a bunch of different forks, such as:

https://github.com/bminor/glibc/tree/master/sunrpc
http://git.linux-nfs.org/?p=steved/nfs-utils.git;a=tree;f=tools/rpcgen;hb=HEAD

> Do you have some hints for me, how can I try this? I don't have any experience with VXI11. e.g. how to generate VXI11 counterpart so I can talk with this example.

VXI11 is a TCP-IP (perhaps with some UDP) based protocol, based on SUN/ONC RPC. It's very similar to USBTMC, and more or less duplicates the functionality of GPIB, but over IP. There is a variant to handle abstracting GPIB interfaces, too. VXI11 is falling out of favor these days, being replaced with HiSLIP which was designed to replace it.

You can use the NI VISA libraries (Or Keysight, or whatnot). I usually use the NI MAX GUI. Connect to a VISA address like "TCPIP::127.0.0.1::INSTR", and it should properly detect the VXI-11 server.

**Comment by j123b567** [comment]
Thank you for your answers. My bad english, I was interested in IDL files not rpcgen itself, but you have answered it anyway, great.

I will merge this even if I can't test it now, because it can be interesting codebase for others.

**Pull request merged**